fuel-core/src/coin_query.rs

```diff
@@ -59,10 +60,19 @@ pub fn largest_first(
 
     for (owner, asset_id, amount) in spend_query {
         let coins_of_asset_id: Vec<(UtxoId, Coin)> = {
-            let coin_ids: Vec<UtxoId> = db
+            let mut coin_ids: Vec<UtxoId> = db
                 .owned_coins_by_asset_id(owner, asset_id, None, None)
                 .try_collect()?;
 
+            // Filter excluded coins
+            coin_ids.retain(|&id| {
+                if let Some(excluded_ids) = excluded_ids {
+                    !excluded_ids.contains(&id)
+                } else {
+                    true
+                }
```

Comment:
[nit] could use unwrap_or instead of if-let-else.
```suggestion
                excluded_ids
                    .map(|excluded_ids| !excluded_ids.contains(&id))
                    .unwrap_or(true)
```